### PR TITLE
Jmprieur/add ciam dev apps

### PR DESCRIPTION
**Pull request by jmprieur**
# Add CIAM samples and fix web API authority
- Adds a ciam web app calling a ciam web API.
- fix the CIAM authority in web APIs.

### CODE REVIEW

File: src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs [hunk 1]
```diff
@@ -253,6 +256,50 @@ public static class MicrosoftIdentityWebApiAuthenticationBuilderExtensions
...
+
+            if (mergedOptions.ExtraQueryParameters != null)
+            {
+                options.MetadataAddress += options.MetadataAddress.Contains('?', StringComparison.OrdinalIgnoreCase) ? "&" : "?";
+                options.MetadataAddress += string.Join("&", mergedOptions.ExtraQueryParameters.Select(p => $"{p.Key}={p.Value}"));
+            }
+        }
+
+        /*
+        private static void UpdateOptionsMetadata(JwtBearerOptions options, MergedOptions mergedOptions)
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs:272]: remove the commented code?

File: tests/DevApps/ciam/newTest.sln [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.0.31903.59
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "myWebApi", "myWebApi\myWebApi.csproj", "{3F20A960-F3AE-4069-9CAC-7F3BD804FA1C}"
+EndProject
```
Comment (jennyf19) [inline tests/DevApps/ciam/newTest.sln:7]: can you give this .sln a better name?

File: tests/DevApps/ciam/myWebApi/myWebApi.csproj [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  <PropertyGroup>
+    <TargetFrameworks>net7.0</TargetFrameworks>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <UserSecretsId>aspnet-myWebApi-17be892d-3bb3-4282-92e2-301b02c86ed6</UserSecretsId>
+    <UseWIP>true</UseWIP>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="7.0.10" />
```
Comment (jennyf19) [inline tests/DevApps/ciam/myWebApi/myWebApi.csproj:12]: >.0.1 [](http://example.com/codeflow?start=71&length=4)

should use the Directory.Build.Props for the test, no?

File: tests/DevApps/ciam/myWebApi/Program.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+using Microsoft.AspNetCore.Authentication;
```
Comment (jennyf19) [inline tests/DevApps/ciam/myWebApi/Program.cs:1]: you should get warnings if the license header is not on the file

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Comment by jennyf19** [comment]
not a fan of adding more dev apps, but I understand why :)

**Comment by jmprieur** [comment]
> not a fan of adding more dev apps, but I understand why :)

We need to catch regressions

**Comment by jmprieur** [comment]
@westin-m, @JoshLozensky anybody else want to review?

File: tests/DevApps/ciam/myWebApi/myWebApi.csproj (hunk 3 above)
Comment (jmprieur) [reply tests/DevApps/ciam/myWebApi/myWebApi.csproj:12]: I'll address this later (with the fact that it's still using IDownstreamWebApi. It's straight from the project templates)

**Pull request merged**